src/dataset/iter.rs

```diff
@@ -77,16 +77,17 @@ where
         if self.target_or_feature && self.dataset.nfeatures() <= self.idx {
             return None;
         }
-
         let mut records = self.dataset.records.view();
         let mut targets = self.dataset.targets.as_targets();
         let feature_names;
+        let mut target_names = vec!["class".to_string()];
         let weights = self.dataset.weights.clone();
 
         if !self.target_or_feature {
             // This branch should only run for 2D targets
             targets.collapse_axis(Axis(1), self.idx);
             feature_names = self.dataset.feature_names.clone();
+            target_names = self.dataset.target_names.clone();
```

Comment:
This branch means we're iterating through the targets, so we should return each target name separately:
```
if self.data.target_names.is_empty() {
  target_names = Vec::new();
} else {
  target_names = vec![self.dataset.target_names[self.idx]];
}
```